src/package/packageVersion.ts

```diff
@@ -217,4 +231,78 @@ export class PackageVersion {
     updateResult.id = await pkgUtils.getSubscriberPackageVersionId(packageVersionId, this.connection);
     return updateResult;
   }
+
+  /**
+   * Increase the wait time for a package version that is org dependent.
+   *
+   * @param resolvedPackageId
+   * @param pollInterval
+   * @param wait
+   * @private
+   */
+  private async resolveOrgDependentPollingTime(
+    resolvedPackageId: string,
+    wait: Duration,
+    pollInterval: Duration
+  ): Promise<Duration> {
+    let w = wait;
+    // If we are polling check to seeif the package is Org-Dependent, if so, update the poll time
+    if (wait.milliseconds > 0) {
+      const query = `SELECT IsOrgDependent FROM Package2 WHERE Id = '${resolvedPackageId}'`;
+      try {
+        const pkgQueryResult = await this.connection.singleRecordQuery<PackagingSObjects.Package2>(query, {
+          tooling: true,
+        });
+        if (pkgQueryResult.IsOrgDependent) {
+          w = Duration.seconds((60 / pollInterval.seconds) * wait.seconds);
+        }
+      } catch {
+        // do nothing
+      }
+    }
+    return w;
+  }
+
+  private async updateProjectWithPackageVersion(
+    withProject: SfProject,
+    results: PackageVersionCreateRequestResult
+  ): Promise<void> {
+    if (withProject && !process.env.SFDX_PROJECT_AUTOUPDATE_DISABLE_FOR_PACKAGE_VERSION_CREATE) {
+      const query = `SELECT Name, Package2Id, MajorVersion, MinorVersion, PatchVersion, BuildNumber, Description, Branch FROM Package2Version WHERE Id = '${results.Package2VersionId}'`;
+      const packageVersion = await this.connection.singleRecordQuery<PackagingSObjects.Package2Version>(query, {
+        tooling: true,
+      });
+      const packageVersionVersionString = `${packageVersion.MajorVersion}.${packageVersion.MinorVersion}.${packageVersion.PatchVersion}.${packageVersion.BuildNumber}`;
+      await this.generatePackageDirectory(packageVersion, withProject, packageVersionVersionString);
+      const newConfig = await generatePackageAliasEntry(
+        this.connection,
+        withProject,
+        packageVersion.SubscriberPackageVersionId,
+        packageVersionVersionString,
+        packageVersion.Branch,
+        packageVersion.Package2Id
+      );
+      withProject.getSfProjectJson().set('packageAliases', newConfig);
+      await withProject.getSfProjectJson().write();
+    }
+  }
+
+  private async generatePackageDirectory(
+    packageVersion: PackagingSObjects.Package2Version,
+    withProject: SfProject,
+    packageVersionVersionString: string
+  ) {
+    const pkg = await (await Package.create({ connection: this.connection })).getPackage(packageVersion.Package2Id);
+    const pkgDir =
```

Comment:
putting some appropriate overloads on the utils for this class might help TS know the return type (should match the T of the first argument or be undefined).

I'm not a fan of `{} as NamedPackageDir` (it's just not).

If `getConfigPackageDirectory` returns an undefined, this method should handle that.